tests/test_report_database.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+from unittest.mock import patch, MagicMock
+from report_app.main.report_database import ReportDatabase
+from moto import mock_dynamodb
+import boto3
+
+
+class TestReportDatabase(unittest.TestCase):
+
+    def setUp(self):
+        self.mock_db = mock_dynamodb()
+        self.mock_db.start()
+
+        boto3.resource('dynamodb', region_name='eu-west-2').create_table(
+            TableName='MOCK_TABLE',
+            KeySchema=[
+                {
+                    'AttributeName': 'name',
+                    'KeyType': 'HASH'
+                }
+            ],
+            AttributeDefinitions=[
+                {
+                    'AttributeName': 'name',
+                    'AttributeType': 'S'
+                }
+            ],
+            ProvisionedThroughput={
+                'ReadCapacityUnits': 5,
+                'WriteCapacityUnits': 5
+            }
+        )
+        self.report_database = ReportDatabase('MOCK_TABLE', 'test_access_key', 'test_secret_key', 'eu-west-2')
+
+    def tearDown(self):
+        self.mock_db.stop()
+
+    def test_init_with_correct_table(self):
+        self.assertEqual(self.report_database._table_name, 'MOCK_TABLE')
+        self.assertIsNotNone(self.report_database._table)
+        self.assertIsNotNone(self.report_database._client)
+
+    def test_init_without_credentials(self):
+        self.assertRaises(ValueError, ReportDatabase, 'MOCK_TABLE', '', '', 'eu-west-2')
+
+    def test_init_with_incorrect_table(self):
+        self.assertRaises(Exception, ReportDatabase, 'INCORRECT_TABLE', 'test_access_key', 'test_secret_key',
+                          'eu-west-2')
+
+    def test_init_without_region(self):
+        self.assertRaises(ValueError, ReportDatabase, 'MOCK_TABLE', 'test_access_key', 'test_secret_key', '')
+
+    @patch('report_app.main.report_database.boto3.resource')
+    @patch('report_app.main.report_database.os.getenv')
+    def test_create_client_with_docker_compose_dev(self, mock_getenv, mock_boto_resource):
+        # Arrange
+        mock_boto_resource.return_value = MagicMock()
+        mock_getenv.return_value = '1'
+        access_key = 'test_access_key'
+        secret_key = 'test_secret_key'
+        region = 'test_region'
+
+        # Act
+        client = ReportDatabase._ReportDatabase__create_client(access_key, secret_key, region)
+
+        # Assert
```

Comment:
I typically omit the comments - but I wonder if the comments are something we should add in to other tests as well to encourage this type of structure?🤔

In my own way, I'm saying I quite like that the comments have been left in here! 😁